sdks/python/apache_beam/coders/coders_test_common.py

```diff
@@ -236,6 +252,13 @@ def test_deterministic_coder(self):
     self.check_coder(deterministic_coder, list(MyIntFlag))
     self.check_coder(deterministic_coder, list(MyFlag))
 
+    self.check_coder(
+        deterministic_coder,
+        [DefinesGetAndSetState(1), DefinesGetAndSetState((1, 2, 3))])
+
+    with self.assertRaises(TypeError):
```

Comment:
Also add a check of trying to use this for a non-deterministic state?

Reply:
good suggestion, done!